python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -348,7 +349,13 @@ def _executeQuantaInProcess(self, graph, butler):
         count, totalCount = 0, len(graph)
         for qnode in graph:
             _LOG.debug("Executing %s", qnode)
-            self.quantumExecutor.execute(qnode.taskDef, qnode.quantum, butler)
+            try:
+                self.quantumExecutor.execute(qnode.taskDef, qnode.quantum, butler)
+            finally:
+                # sqlalchemy has some objects that can last until a garbage
+                # collection cycle is run, which can happen at unpredictable
+                # times, run a collection loop here explicitly.
+                gc.collect()
```

Comment:
Running GC frequently can be very expensive, I wonder if we could avoid pessimization by running it conditionally. In my old project when I did something similar I only ran it when virtual mem size increased significantly since last explicit CG. This of course needs some guessing as to what makes a significant increase, in that project threshold was something like 10MB, but it clearly depends on growth pattern.